### ISSUE TITLE
Agent hanging on fastlane job

### ISSUE THREAD

**Issue by piotrbernad** [buildkite/agent#171]
I have a problem with newer versions than 1.7.0 of fastlane tools. On 1.7.0 everything works fine, but when I try to upgrade fastlane all jobs hanging. Jobs stops on build command. There was a change on this command and now fastlane uses --verbose command as default. Removing this causes that buildkite works fine. 

```
old command:

set -o pipefail && krausefx-ipa build -s "xxx-ios-Production" --ipa "./build/xxx-xxx.ipa" -m "/usr/local/var/buildkite-agent/builds/mac-mini-2.local-1/xxx-2/xxx/InHouse_com.xxx.mobileprovision" | xcpretty

new command:

set -o pipefail && krausefx-ipa build -s "xxx-ios-Production" --ipa "./builds/xxx-xxx.ipa" -m "/usr/local/var/buildkite-agent/builds/mac-mini-2.local-1/xxx-2/xxx/InHouse_com.xxx.mobileprovision" --verbose | xcpretty 
```

**Comment by keithpitt** [comment]
Hey @piotrbernad, thanks for submitting this! We've had a few reports of fastlane hanging. Have you opened a similar issue with them? Maybe you could prod them and see if they know what could be happening?

We haven't anything around logging on our end. It'd be good to hear what they think to help us track down the issue.

**Comment by keithpitt** [comment]
I think maybe @mattdelves hit this problem as well...

**Comment by toolmantim** [comment]
I believe it was @danthorpe who experienced this. Any luck tracking down why Dan?

**Comment by danthorpe** [comment]
Yeah, I bumped into this issue too. However, I never had time to pin it down to the underlying cause. My initial thoughts was that it was something to do with the keychain - as there were lots of related changes happening with Fastlane.

@piotrbernad have you manipulated Fastlane so that it builds without the `--verbose` flag but on versions of `krausefx-ipa` after 1.7.0, and that works?

I haven't raised an issue on https://github.com/KrauseFx/fastlane perhaps Felix @KrauseFx can shed more light?

**Comment by KrauseFx** [comment]
I assume `fastlane` works fine for all of you on your local machine? If so, it might have something to do with when Ruby flushes the text out to the stdout. I assume BuildKite will kill the process if there was no output for some time.

**Comment by danthorpe** [comment]
@KrauseFx does work fine on my local machine. I can't comment on the BK process - @toolmantim?

**Comment by keithpitt** [comment]
@KrauseFx the agent doesn't do any smarts around lack of output for a period of time. It'll just sit there, hanging out, until something happens.

I haven't done iOS in about 2 years, so if someone could quickly whip up an example that's breaking for me, I'll be able to look at it on my machine and try and figure out what's going on.

**Comment by Jasperswaagman** [comment]
I can confirm Fastlane 1.14.x does not work; but 1.7.0 does work. So a workaround would be to use Fastlane v 1.7.0. We didn't try without the `-verbose` flag though.

**Comment by toolmantim** [comment]
I just sent a PR for fastlane to fix the output stream issue, and everything is running nicely in Fastlane 1.15.x. It had to do with the switch from using `pty` to `IO.popen`

**Comment by toolmantim** [comment]
You can use the fixed version in the meantime by switching the github and branch in your application's Gemfile: https://github.com/buildkite/buildkite-fastlane-demo/compare/fix-ipa-build-output

**Comment by toolmantim** [comment]
Another temporary fix is to start your buildkite agent with `--no-pty`

**Comment by toolmantim** [comment]
I also sent a PR for the underlying cause to commander: commander-rb/commander#15 It sets a negative `$terminal.wrap_at` value if the terminal reports it's screen width < 5

**Comment by toolmantim** [comment]
Ok, so what was causing the hang has been fixed in commander-rb/commander#15 and released to rubygems as v4.3.5. Simply `bundle update` and you'll be good to go!

**Comment by danthorpe** [comment]
@toolmantim - fantastic stuff!
